core/src/services/swift/backend.rs

```diff
@@ -261,11 +263,10 @@ impl Accessor for SwiftBackend {
     }
 
     async fn read(&self, path: &str, args: OpRead) -> Result<(RpRead, Self::Reader)> {
-        let range = args.range();
-        let resp = self.core.swift_read(path, range).await?;
+        let resp = self.core.swift_read(path, args).await?;
 
         match resp.status() {
-            StatusCode::OK | StatusCode::PARTIAL_CONTENT => {
+            StatusCode::OK | StatusCode::PARTIAL_CONTENT | StatusCode::RANGE_NOT_SATISFIABLE => {
```

Comment:
This is wrong. We should return empty body for `RANGE_NOT_SATISFIABLE`. Take s3 as reference.

Reply:
Sorry, miss pushed wrong fix commit. Fixed.